broadlink/climate.py

```diff
@@ -238,3 +239,333 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
             input_payload.append(int(weekend[i]["temp"] * 2))
 
         self.send_request(input_payload)
+
+
+class sq1(device):
+    """Controls Tornado SMART X SQ series air conditioners."""
+    def __init__(self, *args, **kwargs):
+        device.__init__(self, *args, **kwargs)
+        self.type = "Tornado SQ air conditioner"
+
+    def _decode(self, response) -> bytes:
+        payload = self.decrypt(bytes(response[0x38:]))
+        return payload
+
+    def _calculate_checksum(self, payload: bytes, byteorder: str) -> bytes:
```

Comment:
If we always return an integer, we can avoid the byteorder parameter.
```python3
    def _calculate_checksum(self, payload: bytes) -> int:
```

Reply:
The get_ac_info request (and possibly others) return the checksum in the opposite byte order to other requests, I thought about having a default, but reckoned keeping it explicit would be more clear, no?

Reply:
The idea is to have a [single responsibility](https://en.wikipedia.org/wiki/Single-responsibility_principle) for this method in order to make it more flexible/reusable. When we ask for a sum, we usually expect an integer. Packing the data for transportation is the responsibility of another layer.